src/qibocal/protocols/readout_characterization.py

```diff
@@ -78,63 +80,55 @@ def _acquisition(
 
     data = ReadoutCharacterizationData(
         qubit_frequencies={
-            qubit: platform.qubits[qubit].drive_frequency for qubit in targets
+            # TODO: should this be the drive frequency instead?
+            qubit: float(platform.calibration.single_qubits[qubit].qubit.frequency_01)
+            for qubit in targets
         },
         delay=float(params.delay),
     )
 
     # FIXME: ADD 1st measurament and post_selection for accurate state preparation ?
 
     for state in [0, 1]:
-        # Define the pulse sequences
-        if state == 1:
-            RX_pulses = {}
-        ro_pulses = {}
         sequence = PulseSequence()
         for qubit in targets:
-            start = 0
+            natives = platform.natives.single_qubit[qubit]
+            ro_channel = natives.MZ()[0][0]
             if state == 1:
-                RX_pulses[qubit] = platform.create_RX_pulse(qubit, start=0)
-                sequence.add(RX_pulses[qubit])
-                start = RX_pulses[qubit].finish
-            ro_pulses[qubit] = []
-            for _ in range(2):
-                ro_pulse = platform.create_qubit_readout_pulse(qubit, start=start)
-                start += ro_pulse.duration + int(
-                    params.delay
-                )  # device required conversion
-                sequence.add(ro_pulse)
-                ro_pulses[qubit].append(ro_pulse)
+                sequence |= natives.RX()
+            sequence |= natives.MZ()
+            sequence.append((ro_channel, Delay(duration=params.delay)))
+            sequence |= natives.MZ()
```

Comment:
I think here we also have the usual problem of qubits not playing in parallel. Since you already have a delay anyway, maybe it is easier to do it manually:
```suggestion
                sequence += natives.RX()
            sequence.append((ro_channel, Delay(duration=natives.RX()[0][1].duration)))
            sequence += natives.MZ()
            sequence.append((ro_channel, Delay(duration=params.delay)))
            sequence += natives.MZ()
```